apps/system/test/unit/nfc_handover_manager_test.js

```diff
@@ -93,14 +86,280 @@ suite('Nfc Handover Manager Functions', function() {
+    teardown(function() {
+      nfcHandoverManager.stop();
+    });
+
+    test('bluetoothStatusSaved is false', function() {
+      nfcHandoverManager.bluetoothStatusSaved = false;
+      this.sinon.stub(Service, 'query').returns(true);
+      nfcHandoverManager.start();
+      nfcHandoverManager._saveBluetoothStatus();
+
+      assert.isTrue(nfcHandoverManager.bluetoothStatusSaved);
+      assert.isFalse(nfcHandoverManager.bluetoothAutoEnabled);
+      assert.ok(Service.query.calledWith('Bluetooth.isEnabled'));
+    });
+
+    test('bluetoothStatusSaved is true', function() {
+      this.sinon.stub(Service, 'query').returns(true);
+      nfcHandoverManager.start();
+      nfcHandoverManager.bluetoothStatusSaved = true;
+      assert.ok(!Service.query.called);
+    });
+  });
+
+  suite('restoreBluetoothStatus', function() {
+    test('if handover is not in progress, bluetooth transfer queue is empty, ' +
+      'and bluetoothAutoEnabled is true', function() {
+        this.sinon.stub(nfcHandoverManager, 'isHandoverInProgress')
+          .returns(false);
+        this.sinon.stub(Service, 'query').returns(true);
+        this.sinon.spy(window, 'dispatchEvent');
+        nfcHandoverManager.bluetoothAutoEnabled = true;
+        nfcHandoverManager._restoreBluetoothStatus();
+
+        assert.isFalse(nfcHandoverManager.bluetoothAutoEnabled);
+        assert.isFalse(nfcHandoverManager.bluetoothStatusSaved);
+        assert.ok(Service.query
+          .calledWith('BluetoothTransfer.isSendFileQueueEmpty'));
+        assert.ok(window.dispatchEvent.called);
+    });
+
+    test('if handover is not in progress, bluetooth transfer queue is empty, '+
+      'and bluetoothAutoEnabled is false', function() {
+        this.sinon.stub(nfcHandoverManager, 'isHandoverInProgress')
+          .returns(false);
+        this.sinon.stub(Service, 'query').returns(true);
+        this.sinon.spy(window, 'dispatchEvent');
+        nfcHandoverManager.bluetoothAutoEnabled = false;
+        nfcHandoverManager._restoreBluetoothStatus();
+
+        assert.isFalse(nfcHandoverManager.bluetoothAutoEnabled);
+        assert.ok(Service.query
+          .calledWith('BluetoothTransfer.isSendFileQueueEmpty'));
+        assert.ok(!window.dispatchEvent.called);
+    });
+
+    test('if handover is not in progress, bluetooth transfer queue ' +
+      'is not empty', function() {
+        this.sinon.stub(nfcHandoverManager, 'isHandoverInProgress')
+          .returns(false);
+        this.sinon.stub(Service, 'query').returns(false);
+        this.sinon.spy(window, 'dispatchEvent');
+        nfcHandoverManager._restoreBluetoothStatus();
+
+        assert.ok(Service.query
+          .calledWith('BluetoothTransfer.isSendFileQueueEmpty'));
+        assert.ok(!window.dispatchEvent.called);
+    });
+
+    test('if handover is in progress, bluetooth transfer queue is empty',
+      function() {
+        this.sinon.stub(nfcHandoverManager, 'isHandoverInProgress')
```

Comment:
Ian in earlier comments pointed out that indents in similar cases should be remove. I don't have a strong opinion on that, but would be good to keep to the same style everywhere.